src/issues/github.rs

```diff
@@ -1,44 +1,79 @@
-use color_eyre::eyre::{eyre, Result};
+use color_eyre::eyre::{eyre, Result, WrapErr};
 
-use crate::app_config::get_or_prompt_for_octocrab;
+use crate::app_config::get_or_prompt_for_github_token;
 use crate::issues::Issue;
 use crate::{config, state};
 
-pub(crate) async fn list_issues(
+const ISSUES_QUERY: &str = r##"
+query($repo: String!, $owner: String!, $labels: [String!]) { 
+  repository(name:$repo, owner:$owner) { 
+    issues(states:OPEN, first: 30, labels: $labels) {
+      nodes {
+        number,
+        title
+      }
+    }
+  }
+}
+"##;
+
+#[derive(serde::Deserialize)]
+struct ResponseIssue {
+    number: usize,
+    title: String,
+}
+
+pub(crate) fn list_issues(
     github_config: Option<config::GitHub>,
     github_state: state::GitHub,
-    labels: Option<Vec<String>>,
+    labels: &Option<Vec<String>>,
```

Comment:
Small nitpick: I believe general recommendation is to use `Option<&T>` rather than `&Option<T>`. Using `Option::as_ref` at the call site should be enough to do this.

Reply:
Is there a reason why one or the other? I think the only reason I changed it from owned was Clippy told me to make things a reference when you didn't need ownership.

Reply:
My guess as to the reason is if the caller as an `&T` they can't easily get to an `&Option<T>` but they could get to an `Option<&T>` which makes sense, so I'll change it. Still curious if there's another reason... also why there isn't a clippy lint for that.

Reply:
Don't know about the clippy lint. I swear I've seen something about it before as a suggestion from either clippy or the compiler. My main though is it's weird to pass a `&None` instead of `None`.